src/JWT.ts

```diff
@@ -485,19 +489,66 @@ export async function verifyJWT(
     throw new Error(`${JWT_ERROR.INVALID_JWT}: No DID has been found in the JWT`)
   }
 
-  const { didResolutionResult, authenticators, issuer }: DIDAuthenticator = await resolveAuthenticator(
-    options.resolver,
-    header.alg,
-    did,
-    proofPurpose
-  )
-  if (didResolutionResult.didDocument?.verificationMethod?.[0].type === 'conditionalProof2022') {
-    return verifyConditionalProof(jwt, didResolutionResult.didDocument.verificationMethod[0])
+  let authenticators: VerificationMethod[]
+  let issuer: string
+  let didResolutionResult: DIDResolutionResult
+  if (options.didAuthenticator) {
+    ;({ didResolutionResult, authenticators, issuer } = options.didAuthenticator)
+  } else {
+    ;({ didResolutionResult, authenticators, issuer } = await resolveAuthenticator(
+      options.resolver,
+      header.alg,
+      did,
+      proofPurpose
+    ))
+    // Add to options object for recursive reference
+    options.didAuthenticator = { didResolutionResult, authenticators, issuer }
+  }
+  console.log(`verifyJWT(): verifying ${did} with ${options.didAuthenticator ? 'provided' : 'resolved'} authenticators`)
+  console.log(authenticators.map((auth) => auth.id).join(', '))
+
+  const { didUrl } = parse(did) as ParsedDID
+
+  let signer: VerificationMethod | null = null
+
+  if (did !== didUrl) {
+    const authenticator = authenticators.find((auth) => auth.id === did)
+    if (!authenticator) {
+      throw new Error(`${JWT_ERROR.INVALID_JWT}: No authenticator found for did URL ${did}`)
+    }
+
+    if (authenticator.type === CONDITIONAL_PROOF_2022) {
```

Comment:
i am saying a logic duplication here can't move the loop to the conditionalAlgorithm file ?